cmd/metal-api/internal/datastore/size.go

```diff
@@ -4,8 +4,61 @@ import (
 	"errors"
 
 	"github.com/metal-stack/metal-api/cmd/metal-api/internal/metal"
+	r "gopkg.in/rethinkdb/rethinkdb-go.v6"
 )
 
+// SizeSearchQuery can be used to search sizes.
+type SizeSearchQuery struct {
+	ID                    *string           `json:"id" optional:"true"`
+	Name                  *string           `json:"name" optional:"true"`
+	Labels                map[string]string `json:"labels" optional:"true"`
+	ReservationsPartition *string           `json:"reservations_partition" optional:"true"`
+	ReservationsProject   *string           `json:"reservations_project" optional:"true"`
+}
```

Comment:
This is hard to get, what about putting these into a separate struct like so:

```suggestion
type SizeSearchQuery struct {
	ID                    *string           `json:"id" optional:"true"`
	Name                  *string           `json:"name" optional:"true"`
	Labels                map[string]string `json:"labels" optional:"true"`
        Reservation *Reservation           `json:"reservation" optional:"true"`
}

type Reservation struct {
     Partition *string
     Project *string
}

```

Reply:
Done.